stdlib/public/core/Optional.swift

```diff
@@ -232,7 +232,7 @@ extension Optional where Wrapped: ~Copyable {
   ) throws(E) -> U? {
     #if compiler(>=6.0) && $NoncopyableGenerics
     switch self {
-    case .some(_borrowing y):
+    case .some(let y):
```

Comment:
I'm assuming these changes are in reaction to warnings about `_borrowing` being deprecated. I had to change these to use the deprecated spellings intentionally in order to ensure that the standard library's `.swiftinterface` is still buildable with the older compiler versions that we support. Until we drop support for those older compiler versions these warnings cannot be addressed I'm afraid.

Reply:
Local testing on release/6.0 indicates we're good -- those old compilers appear to accept this PR (or rather, the sibling PR #73810) as is, with no changes. Switching back from `_borrowing` to `let` may actually be working in our favor here. Lucky escape!